pandas/plotting/_matplotlib/style.py

```diff
@@ -94,3 +74,41 @@ def _maybe_valid_colors(colors):
         colors += colors[:mod]
 
     return colors
+
+
+def _is_cn_color(color: str) -> bool:
+    """Check if color string is CN color, like 'C0', 'C1', etc."""
+    cn_colors = ["C" + str(x) for x in range(10)]
```

Comment:
The default matplotlib color cycle has 10 colors, but if set differently this test won't be valid.